### ISSUE TITLE
New version of PyJWT breaks token_manager.py

### ISSUE THREAD

**Issue by kennethkmp** [PureStorage-OpenConnect/py-pure-client#5]
Looks like upgrading  PyJWT to 2.0.0 breaks the client, I tried downgrading PyJWT back to 1.7.1 and everything works OK.

```
  File "libs/pure1.py", line 112, in get_load
    pure1Client = pure1.Client(private_key_file=pure1_pk_file, private_key_password=pk_pwd, app_id=pure1_api_id)
  File "/usr/local/lib/python3.6/site-packages/pypureclient/pure1/client.py", line 42, in Client
    client = pure1_module.Client(**kwargs)
  File "/usr/local/lib/python3.6/site-packages/pypureclient/pure1/Pure1_1_1/client.py", line 79, in __init__
    payload={'iss': app_id})
  File "/usr/local/lib/python3.6/site-packages/pypureclient/token_manager.py", line 57, in __init__
    self._id_token = self._generate_id_token(headers, payload, private_key)
  File "/usr/local/lib/python3.6/site-packages/pypureclient/token_manager.py", line 64, in _generate_id_token
    return new_jwt.decode()
AttributeError: 'str' object has no attribute 'decode'
```

Setting the app_id as bytes breaks further down.

```
File "libs/pure1.py", line 112, in get_load
    pure1Client = pure1.Client(private_key_file=pure1_pk_file, private_key_password=pk_pwd, app_id=pure1_api_id)
  File "/usr/local/lib/python3.6/site-packages/pypureclient/pure1/client.py", line 42, in Client
    client = pure1_module.Client(**kwargs)
  File "/usr/local/lib/python3.6/site-packages/pypureclient/pure1/Pure1_1_1/client.py", line 79, in __init__
    payload={'iss': app_id})
  File "/usr/local/lib/python3.6/site-packages/pypureclient/token_manager.py", line 57, in __init__
    self._id_token = self._generate_id_token(headers, payload, private_key)
  File "/usr/local/lib/python3.6/site-packages/pypureclient/token_manager.py", line 63, in _generate_id_token
    new_jwt = jwt.encode(payload, private_key, algorithm=self.ALGORITHM, headers=headers)
  File "/usr/local/lib/python3.6/site-packages/jwt/api_jwt.py", line 61, in encode
    cls=json_encoder
  File "/usr/lib64/python3.6/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib64/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib64/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'bytes' is not JSON serializable
```

**Comment by sdodsley** [comment]
Yes - I agree this si a problem - I have raised it internally in Pure to try and get a rsolution.

**Comment by sdodsley** [comment]
There is a typo in the pip installation `setup.py` that is not creating an upper constraint for PyJWT. 
Until we get this resolved, downgrading your PyJWT to 1.7.1 is the workaround

**Comment by lvicainne** [comment]
Hello !
It is possible to update the dependence version to PyJWT 2.0.0 ?
Indeed, for a few days, the current version is raising a CVE : CVE-2022-29217
https://avd.aquasec.com/nvd/2022/cve-2022-29217/
thanks
